src/code/z_bgcheck.c

```diff
@@ -3826,129 +3826,128 @@ u32 SurfaceType_GetData(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId,
 }
 
 /**
- * SurfaceType return CamData Index
+ * SurfaceType Get index of bgCamData
  */
-u32 SurfaceType_GetCamDataIndex(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-
+u32 SurfaceType_GetBgCamDataId(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) & 0xFF;
 }
 
 /**
- * CamData return cameraSType
+ * BgCamData Get setting of bgCamData
  */
-u16 func_80041A4C(CollisionContext* colCtx, u32 camId, s32 bgId) {
-    u16 result;
+u16 SurfaceType_GetBgCamDataSetting(CollisionContext* colCtx, u32 bgCamDataId, s32 bgId) {
+    u16 setting;
     CollisionHeader* colHeader;
-    CamData* camData;
+    CamData* bgCamData;
 
     colHeader = BgCheck_GetCollisionHeader(colCtx, bgId);
     if (colHeader == NULL) {
-        return 0;
+        return CAM_SET_NONE;
     }
-    camData = colHeader->cameraDataList;
-    result = camData[camId].cameraSType;
-    return result;
+    bgCamData = colHeader->bgCamDataList;
+    setting = bgCamData[bgCamDataId].setting;
+    return setting;
 }
 
 /**
- * SurfaceType return cameraSType
+ * SurfaceType Get setting of bgCamData setup
  */
-u16 SurfaceType_GetCameraSType(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+u16 SurfaceType_GetBgCamDataSettingSetup(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
```

Comment:
The difference is that `GetBgCamDataSetting` is the function that actually grabs the setting from the `BgCamData`. What the `setup` does is do a few extra checks to make sure the `BgCamData` is there, and then calls `GetBgCamDataSetting` to get the setting. Thoughts?

Reply:
I think this would we solved by addressing the prefix issue I talk about in another comment with `SurfaceType`, then
`SurfaceType_GetBgCamDataSetting` would be `BgCamData_GetSetting`
and `SurfaceType_GetBgCamDataSettingSetup` -> `SurfaceType_GetBgCamDataSetting`
?